security/subsystem/src/main/java/org/jboss/as/security/service/SimpleSecurityManager.java

```diff
@@ -365,6 +365,26 @@ public void push(final String securityDomain, String userName, char[] password,
         }
     }
 
+    private void propagateSubject(final SecurityContext target, final SecurityContext source) {
+        final SecurityContextUtil previousUtil = source.getUtil();
+        final SecurityContextUtil currentUtil = target.getUtil();
+        if(target.getSecurityDomain() != null && source.getSecurityDomain() != null && target.getSecurityDomain().equals(source.getSecurityDomain())){
+            target.setSubjectInfo(source.getSubjectInfo());
+        } else {
+            currentUtil.createSubjectInfo(previousUtil.getUserPrincipal(), previousUtil.getCredential(), previousUtil.getSubject());
+        }
+        if (previousUtil.getRoles() != null) {
+            try {
+                currentUtil.setRoles((RoleGroup) previousUtil.getRoles().clone());
+            } catch (CloneNotSupportedException e) {
+                // should not happen, RoleGroup supports clone.
+                throw new RuntimeException(e);
+            }
+        }
+        currentUtil.setIdentities(previousUtil.getIdentities(Identity.class));
```

Comment:
I don't know how these identities are now used but this doesn't seem to be something we previously propagated.

Reply:
AFAIR - this is due that SubjectInfo is created, not copied. If you copy, you use identities from copy, upon creation, they are empty.

Reply:
@baranowb I might have misunderstood, but if calling `setIdentities` and `setRoles` is only needed if the `SubjectInfo` is being created, then shouldn't these last two blocks move into the `else` block where `createSubjectInfo` is being called?

Reply:
@fjuma which else block?

Reply:
@baranowb - perhaps @fjuma refers to the `else` block started on line 373 which contains line 374 createSubjectInfo, but I can't say for sure, @fjuma is that the one?

Reply:
That's the one she is referring to. When we discussed this months ago in the Jira I suggested following the original flow when the domains match (the if clause above) and only copy things otherwise. In this case the `setIdentities` and `setRoles` above should have been moved to the else block as @fjuma pointed`. Not sure if @baranowb came across some other issue when doing it but from my perspective the whole thing should be in the else block so that it is only performed when we need to create copies of the subject info.

Reply:
For some reason git displayed commit that did not have it( previous one ). At this moment I cant be sure, let me test( but it seems it should have been in 'else')

Reply:
@fjuma @sguilhen  - yep, tested in 7.3.x ( inclusion of setRoles and setIdentities in 'else') so it makes sense to update this.